overlord/snapstate/snapstate.go

```diff
@@ -355,19 +355,19 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	}
 
 	if flags&skipConfigure != 0 {
+		installSet.MarkEdge(prereq, BeginEdge)
```

Comment:
why not put this just before the if instead of doing this twice?

Reply:
Because we either return installSet, or ts. Ts includes installSet, but edges are not copied from one set to another via AddAll, so they need to be set on both.

Reply:
I see,  no super strong opinion, I note there is a AddAllWithEdges as well

Reply:
Ah, I missed AddAllWithEdges, it's definately batter that way, thanks